File: 알고리즘/Shuffle String.py

class Solution:
    def restoreString(self, s: str, indices: List[int]) -> str:
        di = dict()
        for i in range(len(s)):
            di[indices[i]] = s[i];
        
        a=''
        for i in range(len(s)):
           a+=di[i] 
        return a
    
    def restoreString2(self, s: str, indices: List[int]) -> str:
        A = [''] * len(s)
        for i, c in zip(indices, s):
            A[i] = c
        return "".join(A)
        

File: algorithms/2/Shuffle String.cpp
#include <bits/stdc++.h>

using namespace std;
class Solution {
public:
    string restoreString(string s, vector<int>& indices) {
        unordered_map<int, char> map;
        for(int i=0; i<s.size(); i++){
            map.insert({indices[i], s[i]});
        }
        string temp;
        for(int i=0; i<s.size(); i++){
            temp+=map[i];
        }
        return temp;
    }
    
    string restoreString2(string s, vector<int>& indices) {
        string res = s;
        for(int i =0; i < indices.size(); ++i)
            res[indices[i]] = s[i];
        return res;
    }